modules/Bio/EnsEMBL/ArchiveStableId.pm

```diff
@@ -529,5 +532,11 @@ sub successors {
   return $self->{'successors'};
 }
 
+sub species_id {
+  my $self = shift;
+  $self->{species_id} = shift if (@_);
```

Comment:
We could consider validating the setter value with Bio::EnsEMBL::Utils::Scalar::assert_integer()